src/Oracle.sol

```diff
@@ -43,28 +48,32 @@ contract Oracle is IOracle, DataStreamConsumer, PriceFeedConsumer {
 
     function addRequest(
         address callbackContract,
-        bytes memory callbackArgs
+        bytes memory callbackArgs,
+        uint256 nonce,
+        address sender
     ) external returns (bool) {
         (
             bytes32 id,
             IRequestsManager.RequestStats memory reqStats
-        ) = getRequestProps(callbackContract, callbackArgs);
+        ) = getRequestProps(callbackContract, callbackArgs, nonce, sender);
         // prevent duplicated request execution
-        if (reqStats.status == IRequestsManager.RequestStatus.Fulfilled) {
+        if (reqStats.status == IRequestsManager.RequestStatus.Pending) {
             revert DuplicatedRequestCreation(id);
         }
         requestManager.addRequest(id);
-        emit AutomationTrigger(callbackContract, callbackArgs);
+        emit AutomationTrigger(callbackContract, callbackArgs, nonce, sender);
         return true;
     }
 
     function performUpkeep(bytes calldata performData) external override {
         // Decode the performData bytes passed in by CL Automation.
         // This contains the data returned by your implementation in checkCallback().
-        (bytes[] memory signedReports, bytes memory extraData) = abi.decode(
-            performData,
-            (bytes[], bytes)
-        );
+        (
+            bytes[] memory signedReports,
+            bytes memory extraData,
+            uint256 nonce,
+            address sender
```

Comment:
nonce and sender will be inside `extraData`

Reply:
Fixed